freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -166,7 +165,13 @@ private void handleUser(UmsUsersState.Builder umsUsersStateBuilder, UsersState.B
                     }
                 });
 
-                workloadAdministrationGroups.getWorkloadAdministrationGroupNameList().forEach(groupName -> {
+                // Only add workload admin groups that belong to this environment
+                workloadAdminGroups.stream()
+                        .filter(wag -> wag.getResource().equalsIgnoreCase(environmentAccessRights.getEnvironmentCrn()))
```

Comment:
this doesn't work. WAGs may be associated with resources that are not environments. We don't currently have a mapping of resources to environments, so we can't tell if a WAG is applicable to a given environment.

If we could use the WAG resource, then we could do the filtering on the wags list in `getEnvToUmsUsersStateMap` instead of in `handleUser`

Reply:
So this means we cannot filter at all? I didn't know that there could be other wags, but in this case, it only works for those that CB creates, right?